pkg/controllers/cloud_config_sync_controller_test.go

```diff
@@ -217,6 +233,12 @@ var _ = Describe("Cloud config sync controller", func() {
 		Eventually(
 			apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(infra), infra)),
 		).Should(BeTrue())
+
+		networkResource := makeNetworkResource()
+		Expect(cl.Delete(ctx, networkResource)).To(Succeed())
+		Eventually(
+			apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(networkResource), networkResource)),
```

Comment:
You need to wrap this in an anonymous function else the eventually just receives an `error` and that isn't going to change over time